Add Contribution Section (#20820)
demisto/dockerfiles: README.md

**Note:** We generate nightly information about packages and os dependencies used in each of Demisto's docker images. Checkout the `dockerfiles-info` project [README](https://github.com/demisto/dockerfiles-info/blob/master/README.md) for a full listing.

## Contributing
Contributions are welcome and appreciated.

You can contribute in the following ways:

* [**Create a new Docker Image**](#create-a-new-docker-image) for use in an [XSOAR/XSIAM integration or script](https://xsoar.pan.dev/docs/integrations/docker#scriptintegration-configuration).

> * **Update existing Docker Images:** If you encounter issues or security concerns in our `Dockerfiles`, feel free to open a Pull Request. 
> 
> * **Enhancements:** Feel free to submit Pull Requests with your proposed changes. Enhancements can include optimizations, helper scripts, or improved configurations.
> 
> * **Documentation:** Documentation is crucial. If you find gaps or errors in our documentation, submit PRs to help us improve it. You can also suggest clarifications or additions to make it more user-friendly.

Reviewer: ```suggestion
* **Update existing Docker Images:** to fix issues or security concerns in our `Dockerfiles`. 

* **Enhancements:**  such as optimizations, helper scripts, or improved configurations.

* **Documentation:** Documentation is crucial. If you find gaps or errors in our documentation, please help us improve it. You can suggest clarifications or additions to make it more user-friendly.
```

> ### Licenses

Reviewer: ```suggestion
### CLA Licenses
```

> In order to conntribute, we will need all contributors to sign a contributor license agreement (CLA). By signing a contributor license agreement, we ensure that the community is free to use your contributions.

Reviewer: ```suggestion
All contributors are required to sign a contributor license agreement (CLA) to ensure that the community is free to use your contributions.
```

When opening a new pull request, a bot will evaluate whether you have signed the CLA. If required, the bot will comment on the Pull Request, including a link to accept the agreement. The CLA document is also available for review as a [PDF](https://github.com/demisto/content/blob/master/docs/cla.pdf).
Visit our [Frequently Asked Questions](https://xsoar.pan.dev/docs/concepts/faq#cla-is-pending-even-though-i-signed-the-agreement) article for CLA related issues.

After opening a Pull Request, and in order for the reviewer to understand the context, make sure to link to the corresponding Pull Request from the [Content](https://github.com/demisto/content) repo where this Docker image will be used.

### How to Contribute

#### Prequisites
Make sure you meet the following prerequisites:
* An active GitHub account.

> * A [clone](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository) or a [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) of the repository. The main difference is that if the contribution is done from a fork, the development Docker image created during the Pull Request build will not be deployed to DockerHub whereas opening a Pull Request from a clone will deploy the image.

Reviewer: I think this isnt correct. The only way to contribute is from a fork. Cloning the repository just means to make a local copy, but the commits are done in the same repo.
Contributors arent allowed to write to our repo at all, so currently there isnt a way to deploy their changes to our dockerhub account (we have a task open for this, lets discuss)

The current solution is to follow the build output which provides a command to run in order to load the image locally. I would add this note here

* Python 3, `git`, Docker Engine and [`pipenv`](https://pipenv.pypa.io/en/latest/) or [`poetry`](https://python-poetry.org/docs/#installation) installed locally or in a Codespace.

Once all prerequisites have been met, create a new branch to hold the proposed work:

```bash
git checkout -b my_new_branch
```

### Create a new Docker Image
To create a new Docker Image, you can either:

* **Use the helper script**
  ```bash
  ./docker/create_new_docker_image.py

  usage: create_new_docker_image.py [-h] [-t {python,powershell}]
                                    [-p {two,three}] [-l {alpine,debian,ubuntu}]
                                    [--pkg PKG]
                                    name

  Create a new docker image

>   positional arguments:

Reviewer: can we put this --help block inside a code block?

* **Enhancements:**  such as optimizations, helper scripts, or improved configurations.

* **Documentation:** Documentation is crucial. If you find gaps or errors in our documentation, please help us improve it. You can suggest clarifications or additions to make it more user-friendly.

### CLA Licenses
All contributors are required to sign a contributor license agreement (CLA) to ensure that the community is free to use your contributions.

When opening a new pull request, a bot will evaluate whether you have signed the CLA. If required, the bot will comment on the Pull Request, including a link to accept the agreement. The CLA document is also available for review as a [PDF](https://github.com/demisto/content/blob/master/docs/cla.pdf).
Visit our [Frequently Asked Questions](https://xsoar.pan.dev/docs/concepts/faq#cla-is-pending-even-though-i-signed-the-agreement) article for CLA related issues.

After opening a Pull Request, and in order for the reviewer to understand the context, make sure to link to the corresponding Pull Request from the [Content](https://github.com/demisto/content) repo where this Docker image will be used.

> ### How to Contribute

Reviewer: why have contributing and how to contribute as 2 seperate sections?

Author: Addressed in https://github.com/demisto/dockerfiles/pull/20820/commits/5d530e9e0adcd77c6df52a7cdf1fe3692d1f8d37.

#### Prequisites
Make sure you meet the following prerequisites:
* An active GitHub account.
* A [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) of the repository.
* Python 3, `git`, Docker Engine and [`pipenv`](https://pipenv.pypa.io/en/latest/) or [`poetry`](https://python-poetry.org/docs/#installation) installed locally or in a Codespace.

> Once all prerequisites have been met, create a new branch to hold the proposed work:

Author: https://github.com/demisto/dockerfiles/pull/20820/commits/71ed1cd8363e9a8dce27d26501ea0cab1ff59751

```bash
git checkout -b my_new_branch
```

### Create a new Docker Image
To create a new Docker Image, you can either:

<details>
  <summary><b>Use the helper script <code>docker/create_new_docker_image.py</code></b></summary>

  <pre>
./docker/create_new_docker_image.py

usage: create_new_docker_image.py [-h] [-t {python,powershell}]
                                    [-p {two,three}] [-l {alpine,debian,ubuntu}]
                                    [--pkg PKG]
                                    name

  Create a new docker image

  positional arguments:
    name                  The image name to use without the organization prefix.
                          For example: ldap3. We use kebab-case naming
                          convention.

  optional arguments:
    -h, --help            show this help message and exit
    -t {python,powershell}, --type {python,powershell}
                          Specify type of image to create (default: python)
    -p {two,three}, --python {two,three}
                          Specify python version to use (default: three)

>     -l {alpine,debian,ubuntu}, --linux {alpine,debian,ubuntu}

Author: https://github.com/demisto/dockerfiles/pull/20820/commits/71ed1cd8363e9a8dce27d26501ea0cab1ff59751

* Make sure to commit both `pyproject.toml` and `poetry.lock` files
* To see the locked dependencies run: `poetry export -f requirements.txt --output requirements.txt --without-hashes` 

> ### `pipenv` quick start:

Reviewer: can we add a note in here that poetry is preferred over pipenv?
